java/org/apache/tomcat/util/http/FastHttpDateFormat.java

```diff
@@ -104,8 +104,9 @@ public final class FastHttpDateFormat {
      */
     public static String getCurrentDate() {
         long now = System.currentTimeMillis();
-        // Handle case where time moves backwards (e.g. system time corrected)
-        if (Math.abs(now - currentDateGenerated) > 1000) {
+        // Ignore millisecond part.
+        now -= now % 1000L;
```

Comment:
Is `long now = System.currentTimeMillis() / 1000L;` not faster?

Reply:
Without significant improvement .
1. solution1: Duration: ***322245800ns*** (100000000 times)
```java
    public static String getCurrentDate() {
        long now = System.currentTimeMillis() / 1000L;
        if (now != currentDateGenerated) {
            currentDate = FORMAT_RFC5322.format(new Date(now*1000L));
            currentDateGenerated = now;
        }
        return currentDate;
    }
```
2. solution2: Duration: ***328164000ns*** (100000000 times)
```java
    public static String getCurrentDate() {
        long now = System.currentTimeMillis();
        // Ignore millisecond part.
        now -= now % 1000L;
        if (now != currentDateGenerated) {
            currentDate = FORMAT_RFC5322.format(new Date(now));
            currentDateGenerated = now;
        }
        return currentDate;
    }
```

Reply:
@markt-asf Updated, PTAL